x-pack/plugins/integration_assistant/server/lib/errors/unparseable_csv_error.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaResponseFactory } from '@kbn/core/server';
+import { ErrorThatHandlesItsOwnResponse } from './types';
+import { GenerationErrorCode } from '../../../common/constants';
+import {
+  GenerationErrorAttributes,
+  GenerationErrorBody,
+} from '../../../common/api/generation_error';
+
+const errorCode = GenerationErrorCode.UNPARSEABLE_CSV_DATA;
+
+export interface CSVParseError {
+  message: string[];
+}
+
+export class UnparseableCSVFormatError extends Error implements ErrorThatHandlesItsOwnResponse {
+  attributes: GenerationErrorAttributes;
+
+  constructor(public readonly csvParseErrors: CSVParseError[]) {
```

Comment:
`private readonly csvParseErrors` instead?

Reply:
There is no `csvParseErrors` in `this` here. It is just a local argument in the constructor , meaning it is only accessible inside the constructor definition. It is fine to remove any modifier to this.

Reply:
Right, I did that some time ago but this was not pushed. I think we all agree `csvParseErrors: CSVParseError[]` works best here.